### PR TITLE
spl-token-cli: Add global `--program-id` argument

### PR DESCRIPTION
**Pull request by mvines**
It's desirable to optionally request a different SPL Token program id to `spl-token-cli`. 

Freebie: `spl-token-client` also now requires a program id as input

### CODE REVIEW

File: associated-token-account/program/src/instruction.rs [hunk 1]
```diff
@@ -26,10 +26,25 @@ pub enum AssociatedTokenAccountInstruction {
...
+    create_associated_token_account_with_program_id(
+        funding_address,
+        wallet_address,
+        token_mint_address,
+        &spl_token::id(),
+    )
+}
+
+/// Creates CreateAssociatedTokenAccount instruction
+pub fn create_associated_token_account_with_program_id(
```
Comment (mvines) [inline associated-token-account/program/src/instruction.rs:40]: Or should we just yank ATA 1.0.4 and publish a 1.0.5 with the program_id folded into `create_associated_token_account()`?  1.04 is just 3 days old with probably not a lot of adoption yet

**Comment by mvines** [comment]
PSA: I'll look at `token/rust` next

File: associated-token-account/program/src/instruction.rs (hunk 1 above)
Comment (CriesofCarrots) [reply associated-token-account/program/src/instruction.rs:40]: Yeah, maybe we should. @joncinque and I were pondering making that yank anyway, due to: https://github.com/solana-labs/solana-program-library/issues/2791

**Review by CriesofCarrots — COMMENTED** [review]
Oh nice! Generally a fan of this. I'll give it a close review when conflicts are resolved.

**Comment by mvines** [comment]
> Oh nice! Generally a fan of this. I'll give it a close review when conflicts are resolved.

Dun

> Yeah, maybe we should. @joncinque and I were pondering making that yank anyway, due to: #2791

Cool, I made that change here under the assumption that 1.0.4 is getting yanked

File: token/cli/Cargo.toml [hunk 2]
```diff
@@ -25,7 +25,7 @@ solana-remote-wallet = "=1.9.5"
 solana-sdk = "=1.9.5"
 solana-transaction-status = "=1.9.5"
 spl-token = { version = "3.3", path="../program", features = [ "no-entrypoint" ] }
-spl-associated-token-account = { version = "1.0", path="../../associated-token-account/program", features = [ "no-entrypoint" ] }
+spl-associated-token-account = { version = "1.0.5", path="../../associated-token-account/program", features = [ "no-entrypoint" ] }
```
Comment (CriesofCarrots) [inline token/cli/Cargo.toml:28]: 🙌

**Review by CriesofCarrots — APPROVED** [review]
New conflicts, but otherwise lgtm!

File: token/rust/src/token.rs [hunk 3]
```diff
@@ -172,6 +180,7 @@ where
     #[allow(clippy::too_many_arguments)]
     pub async fn create_mint<'a, S2: Signer>(
         client: Arc<dyn ProgramClient<T>>,
+        program_id: &Pubkey,
```
Comment (CriesofCarrots) [inline token/rust/src/token.rs:183]:
```suggestion
        program_id: &'a Pubkey,
```
Oops, sorry I missed this

**Pull request merged**